### ISSUE TITLE
Custom Classes won't work for protos that serialize to strings

### ISSUE THREAD

**Issue by AThilenius** [dcodeIO/protobuf.js#1099]
protobuf.js version: 6.8.8

TL;DR: Custom Classes won't work for protos that serialize to strings, like google.protobuf.Timestamp because generated code checks `typeof X === 'object'`.

The [proto3 JSON format](https://developers.google.com/protocol-buffers/docs/proto3#json) defines serialized google.protobuf.Timestamp as RFC 3339 strings. This should be possible with [Custom Classes](https://github.com/dcodeIO/protobuf.js#using-custom-classes), unfortunently the generated code for any proto that uses a Timestamp checks if the field is typeof 'object', which it is not. This means that calling `toObject` and `fromObject` on Timestamp itself works (although you have to cast to any in Typescript) but fails when called on a proto that includes a Timestamp.

The generated code causing this looks like:
```js
if (object.createdTimestamp != null) {
    if (typeof object.createdTimestamp !== "object")
        throw TypeError(".chrome.crosinfra.ci.Build.createdTimestamp: object expected");
    message.createdTimestamp = $root.google.protobuf.Timestamp.fromObject(object.createdTimestamp);
}
```

Resulting in the stack trace:
```
/Users/athilenius/local_dev/ci-ui/appengine/build/generated/protos.js:926
                            throw TypeError(".chrome.crosinfra.ci.Build.createdTimestamp: object expected");
                            ^

TypeError: .chrome.crosinfra.ci.Build.createdTimestamp: object expected
    at Function.fromObject (/Users/athilenius/local_dev/ci-ui/appengine/build/generated/protos.js:926:35)
    at Object.<anonymous> (/Users/athilenius/local_dev/ci-ui/appengine/build/index.js:50:48)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:551:12)
    at Function.Module._load (internal/modules/cjs/loader.js:543:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:744:10)
    at startup (internal/bootstrap/node.js:240:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:564:3)
```

**Comment by dcodeIO** [comment]
Closing this issue because it has been inactive for more than two years and the project has changed substantially.

If this is still relevant, please open a fresh issue with current reproduction steps, version details, and any updated context.

Please also consider that maintaining protobuf.js takes time and effort. If the project is valuable to you or your organization, sponsorship is appreciated.

Thanks!
